### ISSUE TITLE
"VS Online preview requires sign-in with Azure identity..." but am already signed in

### ISSUE THREAD

**Issue by Chuxel** [MicrosoftDocs/vsonline#78]
From Twitter: https://twitter.com/code/status/1191791472057757696
----
Can't seem to get #vsonline working...
I'm signed into 
@code
 on my machine and attempting to Register Local Environment (to set up a self hosted environment) but this gives the error "VS Online preview requires sign-in with Azure identity..." (I am already signed-in?) 😐

![image](https://user-images.githubusercontent.com/312252/68243449-5c158600-ffc7-11e9-8f01-609ecddc625c.png)
![image](https://user-images.githubusercontent.com/312252/68243463-6172d080-ffc7-11e9-8dc0-66f648b3f8b8.png)
![image](https://user-images.githubusercontent.com/312252/68243468-65065780-ffc7-11e9-82a8-450cb9773c00.png)

**Comment by Chuxel** [comment]
//cc: @legomushroom

**Comment by osortega** [comment]
cc @taiyosogawa 
This is probably one of the errors we were seeing.

**Comment by taiyosogawa** [comment]
This looks like the issue we've seen when logged into the browser and VS Code with different credentials. @legomushroom, have you seen this?

**Comment by zplume** [comment]
Thanks @Chuxel, I've since tried Signing out of both Azure and VSO and signing in again, and I now see a 401 error.

![image](https://user-images.githubusercontent.com/2255088/68283222-11514800-0073-11ea-9f0b-f66b528f270c.png)

**Comment by jasongin** [comment]
The problem could be that because you signed in with a different identity, you don't have permission to create environments in the Plan created with the other identity. Try creating a new Plan.

(While registering a local environment is free, it does require an association with a billing plan.)

**Comment by zplume** [comment]
OK, going to try again. Thanks for the suggestions!

Here's what I tried on the 'self hosted environment machine':
* Sign Out of VS Online and Azure from VS Code
* Close all instances of VS Code
* Open VS Code
* VS Online: Sign In
* Alert prompt pops up prompting to [Open Link] (https://login.microsoftonline.com...) > click it
* Browser opens, sign-in with Azure subscription account
* "Are you trying to sign in to Visual Studio Services Client?" > [Continue]
* "Open Visual Studio Code?" > [Open Visual Studio Code]
* Azure: Sign In
* Browser opens, sign-in with Azure subscription account
* Switch back to VS Code
* VS Online: Create Plan
* Select Azure subscription
* Name resource group
* Select region
* Name the plan
* VS Online: Register Local Environment
* Name the local environment
* Select the plan (also option to create a new one!)
*  "Local environment registered!" 🎇
* Windows Defender Firewall prompt > Tick the right boxes, [Allow access]

Over on other machine:
* Azure: Sign In
* VS Online: Sign In
* Alert prompt pops up prompting to [Open Link] > click it
* Browser opens, sign-in with Azure subscription account
* "Are you trying to sign in to Visual Studio Services Client?" > [Continue]
* "Open Visual Studio Code?" > [Open Visual Studio Code]
* VS Online: Connect to environment
* Select environment
* "Setting up remote connection: Starting remote server"
* Windows Defender Firewall prompt > Tick the right boxes, [Allow Access]
* **I can see all the files in the self-hosted environment!**  🎉🥳

Thanks for your help!

p.s. there was also a consent prompt at some point but I restarted the process so I could capture all the steps and it didn't reappear.

**Comment by jasongin** [comment]
I'm glad you got it working!

We are working on simplifying the sign-in experience -- eventually you should need to sign in to VS Code only once.

**Comment by TethysSvensson** [comment]
I am having a somewhat similar issue.

I am trying to connect to a vsonline environment in my local vscode running on linux.

I have installed the `ms-vsonline.vsonline` extension.

I first have to sign in the view the available environments. It opens chromium, I sign in, chromium executes `xdg-open`, vscode is refocused, and I am able to choose the environment I want to connect to.

After that point vscode reloads and I have to sign in again. It again opens chromium, I sign in and chromium executes `xdg-open`. However at this point nothing more happens. I have tried to debug it by running `code --open-url 'vscode://ms-vsonline.vsonline/signin?code=...'` manually but nothing seems to happen.
